internal/pkg/aws/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -158,17 +157,48 @@ func initGetLogEventsInput(opts LogEventsOpts) *cloudwatchlogs.GetLogEventsInput
 
 // Example: if the prefixes is []string{"a"} and all is []string{"a", "b", "ab"}
 // then it returns []string{"a", "ab"}.
-func filterStringSliceByPrefix(all, prefixes []string) (res []string) {
-	m := make(map[string]bool)
-	for _, candidate := range all {
-		for _, prefix := range prefixes {
-			if strings.HasPrefix(candidate, prefix) {
-				m[candidate] = true
+func filterStringSliceByPrefix(all, prefixes []string) []string {
+	trie := buildTrie(prefixes)
+	var matches []string
+	for _, str := range all {
+		node := trie
+		for _, char := range str {
+			val, ok := node.children[char]
+			if !ok {
+				break
 			}
+			if val.hasPrefix {
+				matches = append(matches, str)
+				break
+			}
+			node = val
```

Comment:
nit: maybe we can make this a method of `trieNode`?

Reply:
Seconding - trieNode.find seems appropriate